pandas/io/parquet.py

```diff
@@ -163,7 +163,11 @@ def validate_dataframe(df: DataFrame) -> None:
                      each level of the MultiIndex
                     """
                 )
-        elif df.columns.inferred_type not in {"string", "empty"}:
+        elif not df.columns.empty and df.columns.inferred_type not in {
+            "string",
+            "empty",
+        }:
+            # GH 52034: RangeIndex.inferred_dtype is always "integer" if empty
```

Comment:
This might be too broad? An empty Index with int dtype should probably still raise?

E.g. ``Index([1], dtype="int64")`` should behave the same as ``Index([], dtype="int64")``?

Reply:
Hmm actually, do you know why we have this limitation on string columns names? pyarrow doesn't seem to have this limitation.

```
In [25]: tb = pa.Table.from_pandas(pd.DataFrame({1: [2]}))

In [26]: tb
Out[26]:
pyarrow.Table
1: int64
----
1: [[2]]

In [27]: pq.write_table(tb, "abc")

In [28]: pq.read_table("abc")
Out[28]:
pyarrow.Table
1: int64
----
1: [[2]]
```

Reply:
This could be an old artefact (same as for read_orc that we removed a couple of days ago), so I'd be ok with getting rid of this if not necessary